packages/flutter_tools/lib/src/hot.dart

```diff
@@ -249,7 +249,13 @@ class HotRunner extends ResidentRunner {
     }
 
     _observatoryPort = result.observatoryPort;
-    await connectToServiceProtocol(_observatoryPort);
+    try {
```

Comment:
This fixes the error message when we fall over detecting the service protocol port.